### PR TITLE
Love and Peace Potions No Longer Work on Megafauna

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Makes love and pacification potions not work on megafauna.

## Why It's Good For The Game

It is in line with other megafauna mechanics. You can't date Bubblegum. You just can't.

## Changelog
:cl:
balance: Love and pacification potions no longer work on megafauna.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 1]
```diff
@@ -43,7 +43,8 @@
 	if(internal_type && true_spawn)
 		internal = new internal_type(src)
 	apply_status_effect(STATUS_EFFECT_CRUSHERDAMAGETRACKING)
-	add_trait(TRAIT_NO_TELEPORT, MEGAFAUNA_TRAIT)
+	add_trait(TRAIT_NO_TELEPORT)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: revert this

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/traits.dm#L144

Should be clear in the define file that it can act as both a trait and a source.

**Comment by Shadowflame909** [comment]
Well at least love magic works on lesser  slaughter demons

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 1 above)
Comment (nemvar) [reply code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: Tested it locally. Did not work without this change for some unknown reason.

**Comment by HomocidlePancakes** [comment]
Eh? Didn't we already cuck the timestop so this is no longer possible?

**Comment by nemvar** [comment]
Xenobio gets a crossbreed that removes mob aggro

**Comment by HomocidlePancakes** [comment]
Cuck that instead of the love potion. I wanna see how will people will come up with a workaround for it

**Comment by ExcessiveUseOfCobblestone** [comment]
I don't want the interaction to be possible unless it's part of the boss' gimmick.

**Comment by HomocidlePancakes** [comment]
Since the timestop nerf it's not possible anyway so just... Why?

**Comment by nemvar** [comment]
You can just tank a few hits if you really need to. Bos

**Comment by ExcessiveUseOfCobblestone** [comment]
> Since the timestop nerf it's not possible anyway so just... Why?

Because the possibility would still exist and as new content gets added this interaction may once again be plausible. This PR hits it at the source.

I don't see how that has any bearing to begin with tbh.

**Comment by fludd12** [comment]
I intended to add this in the first place, but forgot about megafauna entirely.

**Comment by ExcessiveUseOfCobblestone** [comment]
unforgivable.

The toll for this transgression is another chapter of crossbreeding, chop chop

**Comment by Daxxed** [comment]
I thought departmental cooperation was heavily rewarded as it requires ALOT of effort OR trust.
This takes a while to pull off, making bubs love you also doesn't magically put him on station, theres more to it.

**Comment by wesoda25** [comment]
Nah this shit is stupid and everyone downvoting doesn’t know how strong bubs really is.

Crazy how so many science players think that since they put 40 minutes of work in, they’re entitled to the most overpowered gear in the game.

**Comment by Whoneedspacee** [comment]
40 minutes is a pretty damn long time to be fairly honest. But yeah megafauna loot of this caliber should not come without risk.

**Comment by HomocidlePancakes** [comment]
>You can just tank a few hits if you really need to. Bos

This is only possible with the hirophant, but being able to cheese it by tanking it with a crusher/saw is more of a design flaw with the boss.

>the possibility would still exist and as new content gets added this interaction may once again be plausible

Anything can happen. I might win a lottery or die from some obscure disease tommorow. We all understand it's not going to happen BUT it's "plausible". Now how actually "plausible" is this interaction coming back in a way that will let you consistently pull it off with no risk at all like it was?

I think it would be better for the game if we keep interactions like this and instead gut the ways to achive them if they become to easy to abuse. Makes the game alot more deep and fun for the people who come up with creative solutions on how to achieve them.

**Comment by 4dplanner** [comment]
I don't see why you need MEGAFAUNA_TRAIT as opposed to a simple istype(mob/living/simple_animal/hostile/megafauna)

**Comment by 81Denton** [comment]
>I think it would be better for the game if we keep interactions like this

I agree with you, absurd things like this are what make SS13 fun. The issue is that there are a ton of ways to cheese mob AI though.
All you need to do is make them stand still for a few second and voila, you can just beat them to death without retaliation.

**Comment by 4dplanner** [comment]
I meant istype(M, /mob/living/simple_animal/hostile/megafauna), sorry.

**Comment by MadmanMartian** [comment]
My suggestion: Rather than an istype, look for a flag. Easier to maintain.

On /vg/ we've got IS_PACIFIABLE for a similar feature.

**Comment by ExcessiveUseOfCobblestone** [comment]
> I think it would be better for the game if we keep interactions like this and instead gut the ways to achive them if they become to easy to abuse. Makes the game alot more deep and fun for the people who come up with creative solutions on how to achieve them.

It's not the source that I dislike, IE timestopping bosses and spamming KA shots sounds like a wacky and cool approach to overcome the boss. Balance wise, It's also neat in the fact it can backfire if you're not careful so you have to make sure you time it appropriately to get the most out of each disable. I'd say you have successfully prepped for the boss fight.

The fact you can entirely disable a boss from fighting you 100-0 with a single, nonmaintainence item is absurd, even if it takes "40 minutes" to get and has a small delay to apply. All it takes is for 1 person to figure out how to disable the boss long enough to apply it and tada now I gotta nerf something else rather than hit this particular interaction.

For someone arguing about the sandbox nature of the game, i'm a bit confused why you want to take the approach of limiting every other item's general interaction to ensure this interaction isn't possible rather than just disabling a single, specific one while enabling everything else.

**Pull request merged**